eth_interface.md

```diff
@@ -19,11 +19,13 @@ We also define the following WebAssembly data types:
 
 ## useGas
 
-Subtracts an amount to the gas counter
+Reduces the gas left counter by an amount.
+
+It should also charge for memory cost by multiplying `memory_pages * memory_cost * amount`, where `memory_cost` is defined by the gas schedule.
```

Comment:
Actually this could be implemented in the metering process by
```
(call $useGas
  (i64.mul
    (i64.const <instruction gas>)
    (i64.mul (i64.const <memory cost per page>) (current_memory))))
```

Though the two `mul`s could be precomputed into one. I'd say this wouldn't have any lower overhead then doing it in the interface, but does remove possibility to adjust memory cost **after** metering.

Reply:
1. Shouldn't it just charge for "added" memory pages instead of charging for all of them each time?
2. In the comment you have `<instruction gas> * <memory cost>`. Shouldn't it be `+`?

Reply:
The idea here is to charge for memory pages for the given time those are alive. The time dimension is metered by `instruction gas` (eg how many cycles it takes). In every block the entire available memory is charged for the duration it is required.

This is an idea @wanderer and I discussed and removes the need for the quadratic memory cost rule.

Reply:
Yes, I noticed it will be some-way quadratic.